recipes/tixi3/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools import files
+from conan import ConanFile
+import os
+import textwrap
+
+required_conan_version = ">=1.45.0"
+
+
+class Tixi3Conan(ConanFile):
+    name = "tixi3"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A simple xml interface based on libxml2 and libxslt"
+    topics = ("tixi3", "xml")
+    homepage = "https://github.com/DLR-SC/tixi"
+    license = "Apache-2.0"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def requirements(self):
+        self.requires("libxml2/2.9.14")
+        self.requires("libxslt/1.1.34")
+        self.requires("libcurl/7.84.0")
+
+    def layout(self):
+        cmake_layout(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+        # tixi is a c library
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            files.copy(self, patch["patch_file"], src=self.recipe_folder, dst=self.export_sources_folder)
+
+    def source(self):
+        files.get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
+
+    def build(self):
+        files.apply_conandata_patches(self)
+
+        cmake = CMake(self)
+        cmake.configure(variables={
+            "TIXI_BUILD_EXAMPLES": "OFF"
+        })
```

Comment:
```suggestion
        cmake.configure()
```
Variables should be declared in the toolchain. More info: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-generate-method